### PR TITLE
feat: add sign in and sign out components

### LINKED ISSUE #1: Set up Sign In and Sign Up components

### PR DESCRIPTION
**Pull request by pyphilia**
This PR contains the component and related files for Signing In and Signing Up.

close #1

### CODE REVIEW

File: .huskyrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "hooks": {
+    "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
+    "post-checkout": "yarn install",
+    "pre-commit": "pretty-quick --staged && yarn lint",
+    "post-commit": "git status",
+    "post-merge": "yarn install",
+    "pre-push": "yarn lint && yarn build"
```
Comment (juancarlosfarah) [inline .huskyrc:8]: Here we should add `yarn test` eventually.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    "node-fetch": "2.6.1",
+    "prop-types": "15.7.2",
+    "qs": "6.10.1",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-i18next": "11.8.5",
+    "react-router": "5.2.0",
+    "react-router-dom": "5.2.0",
+    "react-scripts": "4.0.1",
+    "react-toastify": "^7.0.3"
```
Comment (juancarlosfarah) [inline package.json:20]: Pin

File: package.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    "@material-ui/core": "4.11.2",
+    "@material-ui/icons": "4.11.2",
+    "@material-ui/lab": "4.0.0-alpha.57",
+    "history": "5.0.0",
+    "i18next": "19.8.4",
+    "node-fetch": "2.6.1",
+    "prop-types": "15.7.2",
+    "qs": "6.10.1",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
```
Comment (juancarlosfarah) [inline package.json:15]: Pin

File: package.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "@commitlint/cli": "11.0.0",
+    "@commitlint/config-conventional": "11.0.0",
+    "@cypress/code-coverage": "3.9.2",
+    "@cypress/instrument-cra": "1.4.0",
+    "@testing-library/jest-dom": "^5.11.6",
```
Comment (juancarlosfarah) [inline package.json:59]: Pin

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (juancarlosfarah) [inline public/index.html:10]: Change to "Authentication portal for the Graasp ecosystem."

File: public/manifest.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (juancarlosfarah) [inline public/manifest.json:3]: Update

File: src/types/member.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export const FLAG_GETTING_CURRENT_MEMBER = 'FLAG_GETTING_CURRENT_MEMBER';
```
Comment (juancarlosfarah) [inline src/types/member.js:1]: Not for now, but let's capture the idea of using Redux Saga Routine here.

File: src/config/paths.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export const buildSignInPath = (to) => {
```
Comment (juancarlosfarah) [inline src/config/paths.js:1]: Why not use the `qs` package?

File: src/config/messages.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+export const SIGN_IN_ERROR_MESSAGE = 'There was an error while signing in.';
+export const SIGN_IN_SUCCESS_MESSAGE =
+  'You successfully signed in. You will receive a mail.';
```
Comment (juancarlosfarah) [inline src/config/messages.js:3]: `an email`

File: src/config/messages.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+export const SIGN_UP_ERROR_MESSAGE = 'There was an error signing up.';
+export const SIGN_UP_SUCCESS_MESSAGE =
+  'You successfully signed up. You will receive a mail.';
```
Comment (juancarlosfarah) [inline src/config/messages.js:6]: `an email`

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Nice PR @pyphilia! A couple of minor comments and suggestions. One thing to note, though, is that we should probably include some sort of loading mechanism while the `getCurrentMember` call is going on. Maybe a the small MUI loading circle in the Sign In/Up button (and the button disabled). What do you think?

**Comment by pyphilia** [comment]
I totally agree. Until now I let the activity/loading feedback aside, I think it will greatly rely on react-query if we use it. The layout will also probably change so I think we can let such visual feedback for later.

**Pull request merged**